--- a/onnxruntime/core/session/environment.cc
+++ b/onnxruntime/core/session/environment.cc
@@ -288,15 +288,17 @@
 #ifdef USE_DML
       dml::RegisterDmlSchemas();
 #endif
-      RegisterOnnxOperatorSetSchema();
+      if (IsOnnxStaticRegistrationDisabled()) {
+        RegisterOnnxOperatorSetSchema();
 
 #ifndef DISABLE_ML_OPS
-      RegisterOnnxMLOperatorSetSchema();
+        RegisterOnnxMLOperatorSetSchema();
 #endif
 
 #if defined(ENABLE_TRAINING_OPS)
-      RegisterOnnxTrainingOperatorSetSchema();
-#endif
+        RegisterOnnxTrainingOperatorSetSchema();
+#endif
+      }
 
 #if defined(ENABLE_TRAINING_OPS)
       // preserve this order until <training schemas>: this depends on operatorsetschema registration.
